### DISCUSSION TITLE
Derivative with Respect to Outputs in DeepXDE to Train PINN.

### DISCUSSION THREAD

**Discussion by inodozi** [lululxvi/deepxde#1468] [Q&A]
I'm working with DeepXDE to solve a set of coupled PDEs. My model has three outputs $(y_1, y_2, y_3)$ and two inputs $(x,t)$, with given boundary conditions. Here's the thing: In my PDEs, I need to take the derivative with respect to one of the outputs. For example:

$\frac{\partial y_1}{\partial t}=f_1(x,t)$

$\frac{\partial y_2}{\partial t}=f_2(x,t)+\nabla_{y_1} G_2(y_1,x,t)$

$\frac{\partial y_3}{\partial t}=f_3(x,t)+\nabla_{y_3} G_3(y_3,x,t)$

I know with standard feedforward NNs, we usually take derivatives w.r.t inputs. But how do I do this with an output for PINN training in DeepXDE?

Would appreciate any guidance!

**Comment by lululxvi** [comment]
It is the same.
